docs/contributing/unit-test-guideline.md

```diff
@@ -0,0 +1,193 @@
+A boundary test is testing values on or near the boundaries of allowed inputs.
+
+```go
+func AgeTest(val int32) (ok bool) {
+	if val >= 20 && val < 65 {
+		ok = true
+	}
+	return
+}
+```
+
+When the above function is given as target, the minimum number of tests based on the "Robust Boundary" test is 7.
+  - the input `val` cases: 19, 20, 21, 30, 64, 65, 66
+  
+The robust boundary test requires 6N+1 test cases (N is the number of inputs for target function).
+More increasing the required input, more increasing the minimum number of cases.
+Sometimes, it requires a lot of resources to create/maintain test.
+So, you may test the critical cases using boundary value and have a sense of purpose.
+
+#### Equivalence Class Testing
+
+Equivalence Class Testing is grouping the values together such that all members of the group are considered the same and testing one value from each.
+```go
+func AgeTest(val int32) (ok bool) {
+	if val >= 20 && val < 65 {
+		ok = true
+	}
+	return
+}
+```
+
+When the above function is given as target, you have to create 3 groups and pick one value from each group.
+
+|Desc|Input range|Chose Input|
+|:-----:|:-----:|:-----:|
+| Under 20    | {0, ...,  19 }| 5  |
+| worker range| {20, ..., 64} | 29 |
+| Senior      | {65, ...    } | 72 |
+
+### Vald Style
+
+In the Vald, we create unit tests based on the basic test case.
+And, you also create unit tests based on robust boundary tests or equivalence class tests as needed.
+
+But, we have to take care about that the Vald is developed using Go.
+As you know as Go has many coding features as other languages.
+One of the features is the Go will convert a single value to a slice value when the Function or Method receives variadic argument (e.g. `...[]int`, `...[]string`, `...interface{}`, or etc) as the input.
+It is clear for Go coders. 
+
+And we apply the table-driven test for running unit tests.
+For example, when we create the unit test of `func getMeta(...[]int)`, the test code will be more complex than other functions' test which don't use variadic argument as the input if we create the test for all input patterns.
+Considering those, we define the basic unit case which is a little diffrent from [the basic test case](#Basic).
+
+This change is very clear and you can apply it easily.
+Our basic test case depends on the type of 2 variadic argument.
+
+  1. When input is `...interface{}`
+      - we have to all test cases with value which satisfies `...interface{}` as same as [basic test case](#Basic). For example, `val = 1`, `val = "input"`, `val = []float64{2020.12}` and so on.
+  1. When input is not `...interface{}` but `...[]int`, `...[]string` or etc
+      - we have to create only slice pattern test cases, which is the same as not create test cases with a single vale.
+      - we should test with boundary cases, for example, we should test with `val = []int{math.MaxInt64()}` when the input value is `...[]int`.
```

Comment:
not sure if i am correct but the difference here is testing `interface{}` vs other types

Reply:
🤔 
this section mentions about variadic argument